bindings/ruby/lib/fdbtuple.rb

```diff
@@ -117,12 +117,28 @@ def self.decode(v, pos)
       elsif code == @@STRING_CODE
         epos = find_terminator(v, pos+1)
         [v.slice(pos+1, epos-pos-1).gsub("\x00\xFF", "\x00").force_encoding("UTF-8"), epos+1]
-      elsif code >= @@INT_ZERO_CODE && code <= @@POS_INT_END
+      elsif code >= @@INT_ZERO_CODE && code < @@POS_INT_END
         n = code - @@INT_ZERO_CODE
         [("\x00" * (8-n) + v.slice(pos+1, n)).unpack("Q>")[0], pos+n+1]
-      elsif code >= @@NEG_INT_START and code < @@INT_ZERO_CODE
+      elsif code > @@NEG_INT_START and code < @@INT_ZERO_CODE
         n = @@INT_ZERO_CODE - code
         [("\x00" * (8-n) + v.slice(pos+1, n)).unpack("Q>")[0]-@@size_limits[n], pos+n+1]
+      elsif code == @@POS_INT_END
+        length = v.getbyte(pos+1)
+        val = 0
+        length.times do |i|
+          val = val << 8
+          val += v.getbyte(pos+2+i)
+        end
+        [val, pos+length+2]
+      elsif code == @@NEG_INT_START
+        length = v.getbyte(pos+1) ^ 0xff
+        val = 0
+        length.times do |i|
+          val = val << 8
+          val += v.getbyte(pos+2+i)
```

Comment:
Hm, I can't find a better way of doing this, though my ruby knowledge isn't great. It seems less than ideal that we have to roll our own "turn this byte array into an `Integer`" logic.

Reply:
I was unable to find a better way to do this when I looked myself. We can also change this at any time if we find a better way.